drivers/soundwire/bus.c

```diff
@@ -1310,7 +1312,7 @@ static int sdw_handle_dp0_interrupt(struct sdw_slave *slave, u8 *slave_status)
 			*slave_status = clear;
 		}
 
-		/* clear the interrupt */
+		/* clear the interrupts but don't touch reserved and SDCA_CASCADE fields */
 		ret = sdw_write(slave, SDW_DP0_INT, clear);
```

Comment:
ok, yes, now this won't clear any unhandled interrupts. I haven't read this register description, so, *maybe* this is correct, but it's also possible and also somewhat likely, that this whole register will be clear-on-set. Sometimes good documentation says something like "bitN: reserved - always write 0 to it" or "writes ignored" or something else. But in fact, as long as we don't handle those interrupts, just clearing them shouldn't be a very bad idea either - they won't interrupt again then at least...

Reply:
SDCA_CASCADE is read-only and not clearable per spec.